src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,567 @@
+            outbound_connectivity_check, storage_space_available = check_cluster_outbound_connectivity(diagnoser_container_log, filepath_with_timestamp, storage_space_available)
+
+        if(dns_check and outbound_connectivity_check):
+            return True, storage_space_available
+        return False, storage_space_available
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to perform diagnoser continer check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return True, storage_space_available
+
+
+def check_cluster_DNS(diagnoser_container_log, filepath_with_timestamp, storage_space_available):
+
+    try:
+        # To retreive only the DNS lookup result from the diagnoser container logs
+        dns_check = diagnoser_container_log[0:len(diagnoser_container_log) - 5:]
+
+        # Validating if DNS is working or not and displaying proper result
+        if("NXDOMAIN" in dns_check or "connection timed out" in dns_check):
+            print("Error: We found an issue with the DNS resolution on your cluster. For details about debugging DNS issues visit 'https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/'.\n")
+            if storage_space_available :
+                dns_check_path = os.path.join(filepath_with_timestamp, "DNS_Check.txt")
+                with open(dns_check_path, 'w+') as dns:
+                    dns.write(dns_check + "\nWe found an issue with the DNS resolution on your cluster.")
+            return False, storage_space_available
+
+        else:
+            if storage_space_available :
+                dns_check_path = os.path.join(filepath_with_timestamp, "DNS_Check.txt")
+                with open(dns_check_path, 'w+') as dns:
+                    dns.write(dns_check + "\nCluster DNS check passed successfully.")
+            return True, storage_space_available
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+
+    except Exception as e:
+        logger.warning("An expection has occured while performing the DNS check on the cluster. Exception: {}".format(str(e)) + "\n")
+    
+    return True, storage_space_available
+
+
+def check_cluster_outbound_connectivity(diagnoser_container_log, filepath_with_timestamp, storage_space_available):
+
+    try:
+
+        # To retreive only the outbound connectivity result from the diagnoser container logs
+        outbound_check = diagnoser_container_log[-4:-1:]
+
+        # Validating if outbound connectiivty is working or not and displaying proper result
+        if(outbound_check != "000"):
+            if storage_space_available :
+                outbound_connectivity_check_path = os.path.join(filepath_with_timestamp, "Outbound_Network_Connectivity_Check.txt")
```

Comment:
what url do we test for this check?

Reply:
Will raise the PR for the script and dockerfile today itself so that you get more clear idea.